### PR TITLE
Lock hearing day | Error handling

### LINKED ISSUE #8195: Lock hearing day | Error handling
If a user tries to lock a VACOLS hearing day, the server will return an error. We should catch that error on the frontend and display a message that VACOLS hearing days can't be locked.

1) Update lockHearingDay in DailyDocketContainer to catch errors
1) Add a prop for onErrorHearingDayLock
1) Display an alert when onErrorHearingDayLock is true

### PR DESCRIPTION
**Pull request by Sjones352**
Resolves #8195

### Description
This pr shows an error you try to lock a hearing day in VACOLS.

### Acceptance Criteria
1) Update lockHearingDay in DailyDocketContainer to catch errors
1) Add a prop for onErrorHearingDayLock
1) Display an alert when onErrorHearingDayLock is true

### Testing Plan
1. Go to Daily Docket and attempt to lock a hearing day

screen shot 2018-12-18 at 1 30 08 pm

### CODE REVIEW

File: client/app/hearingSchedule/components/DailyDocket.jsx [hunk 1]
```diff
@@ -389,9 +389,18 @@ export default class DailyDocket extends React.Component {
...
+        title={` Unable to delete Hearing Day
                 ${moment(this.props.dailyDocket.hearingDate).format('M/DD/YYYY')} in Caseflow.`}
         message="Please delete the hearing day through VACOLS" />}
+
+      { this.props.onErrorHearingDayLock && <Alert
+        type="error"
+        styling={alertStyling}
+        title={`VACOLS Hearing Day ${moment(this.props.dailyDocket.hearingDate).format('M/DD/YYYY')}
+           cannot be locked in Caseflow.`}
+        message="Please unlock the hearing day through VACOLS"
```
Comment (sharonwarner) [inline client/app/hearingSchedule/components/DailyDocket.jsx]: I don't think hearing days can be locked in VACOLS, let's just say 'VACOLS hearing days cannot be locked'. cc @laurenrussell

Comment (laurenrussell) [reply client/app/hearingSchedule/components/DailyDocket.jsx]: @Sjones352 @sharonwarner This makes sense. My understanding is the same, that only Caseflow has lock hearing day functionality. If its a VACOLS Hearing Day, and it cannot be locked, let's say '**VACOLS Hearing Day cannot be locked**'

Comment (Sjones352) [reply client/app/hearingSchedule/components/DailyDocket.jsx]: Fixed. @laurenrussell please see screenshot above.

File: client/app/hearingSchedule/components/DailyDocket.jsx [hunk 2]
```diff
@@ -392,6 +392,15 @@ export default class DailyDocket extends React.Component {
         title={` Unable to delete Hearing Day 
                 ${moment(this.props.dailyDocket.hearingDate).format('M/DD/YYYY')} in Caseflow.`}
         message="Please delete the hearing day through VACOLS" />}
+
+      { this.props.onErrorHearingDayLock && <Alert
+        type="error"
+        styling={alertStyling}
+        title={` Unable to delete Hearing Day
```
[the remark this answers is not in the document]
Comment (Sjones352) [reply client/app/hearingSchedule/components/DailyDocket.jsx]: Fixed

File: client/app/hearingSchedule/components/DailyDocket.jsx [hunk 3]
```diff
@@ -389,9 +389,18 @@ export default class DailyDocket extends React.Component {
...
                 ${moment(this.props.dailyDocket.hearingDate).format('M/DD/YYYY')} in Caseflow.`}
         message="Please delete the hearing day through VACOLS" />}
+
+      { this.props.onErrorHearingDayLock && <Alert
+        type="error"
+        styling={alertStyling}
+        title={`VACOLS Hearing Day ${moment(this.props.dailyDocket.hearingDate).format('M/DD/YYYY')}
+           cannot be locked in Caseflow.`}
+        message="VACOLS Hearing Day cannot be locked"
```
Comment (sharonwarner) [inline client/app/hearingSchedule/components/DailyDocket.jsx]: 'Day' should be plural. So this should read 'VACOLS hearing days cannot be locked".

**Pull request merged**